autoPyTorch/pipeline/components/preprocessing/tabular_preprocessing/feature_preprocessing/PCA.py

```diff
@@ -0,0 +1,69 @@
+from typing import Any, Dict, Optional
+
+from ConfigSpace.configuration_space import ConfigurationSpace
+from ConfigSpace.hyperparameters import (
+    CategoricalHyperparameter,
+    UniformFloatHyperparameter,
+)
+
+import numpy as np
+
+import sklearn.decomposition
+from sklearn.base import BaseEstimator
+
+from autoPyTorch.datasets.base_dataset import BaseDatasetPropertiesType
+from autoPyTorch.pipeline.components.preprocessing.tabular_preprocessing.feature_preprocessing. \
+    base_feature_preprocessor import autoPyTorchFeaturePreprocessingComponent
+from autoPyTorch.utils.common import FitRequirement, HyperparameterSearchSpace, add_hyperparameter
+
+
+class PCA(autoPyTorchFeaturePreprocessingComponent):
+    def __init__(self, keep_variance: float = 0.9999,
```

Comment:
The citation from the doc
keep_variance:
select the number of components such that the amount of variance that needs to be explained is greater than the percentage specified by n_components